src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,159 @@
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node node = new Node(tail, value, null);
+        if (isEmpty()) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!(checkIndex(index) || index == size)) {
+            throw new IndexOutOfBoundsException("Failed add to LinkedList. Please check index");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node currentNode = findIndex(index);
+        Node newNode = new Node(currentNode.previous, value, currentNode);
+        if (index == 0) {
+            head = newNode;
+        } else {
+            currentNode.previous.next = newNode;
+        }
+        currentNode.previous = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed get from LinkedList. Please check index");
+        }
+        return findIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed set in LinkedList. Please check index");
+        }
+        Node node = findIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (!checkIndex(index)) {
+            throw new IndexOutOfBoundsException("Failed delete from LinkedList."
+                    + " Please check index");
+        }
+        Node node = findIndex(index);
+        if (node == tail && node == head) {
+            head = null;
+            tail = null;
+        } else if (node == tail) {
+            node.previous.next = null;
+            tail = node.previous;
+        } else if (node == head) {
+            node.next.previous = null;
+            head = node.next;
+        } else {
+            node.previous.next = node.next;
+            node.next.previous = node.previous;
+        }
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (isEmpty()) {
+            throw new IndexOutOfBoundsException("Cannot delete from empty list");
+        }
+        int index = getIndex(object);
+        if (index == -1) {
+            return false;
+        }
+        remove(index);
```

Comment:
you search index by value and call remove(index). In remove you search node by index, so that's 2 loops instead of 1